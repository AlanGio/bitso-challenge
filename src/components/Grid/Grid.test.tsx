import React from 'react';
import { render } from '@testing-library/react';
import { Grid } from './index';

describe('<Grid />', () => {

  it('should render Grid component with 4 rows, 4 columns and islands empty', () => {

    const mockOnlySeaMap = [
      [false, false, false, false], [false, false, false, false],
      [false, false, false, false], [false, false, false, false]
    ];

    const { container } = render(
      <Grid cells={mockOnlySeaMap} onClickCell={() => {}} />
    );

    // Check if Grid hasn't any island
    expect(container.querySelectorAll('.cell.land.col').length).toBe(0);

    // Check if has the right rows length
    expect(container.querySelectorAll('.cell-row').length).toBe(mockOnlySeaMap.length);

    // Check if has the right columns length
    expect(container.querySelectorAll('.cell-row:first-child > .cell').length).toBe(mockOnlySeaMap[0].length);

    expect(container).toMatchSnapshot();

  });

});
