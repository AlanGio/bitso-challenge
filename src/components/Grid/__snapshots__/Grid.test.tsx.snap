// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> should render Grid component with 4 rows, 4 columns and islands empty 1`] = `
<div>
  <div
    class="component-grid"
  >
    <div
      class="grid-container container-fluid"
    >
      <div
        class="cell-row row"
      >
        <div
          class="cell col"
        />
        <div
          class="cell col"
        />
        <div
          class="cell col"
        />
        <div
          class="cell col"
        />
      </div>
      <div
        class="cell-row row"
      >
        <div
          class="cell col"
        />
        <div
          class="cell col"
        />
        <div
          class="cell col"
        />
        <div
          class="cell col"
        />
      </div>
      <div
        class="cell-row row"
      >
        <div
          class="cell col"
        />
        <div
          class="cell col"
        />
        <div
          class="cell col"
        />
        <div
          class="cell col"
        />
      </div>
      <div
        class="cell-row row"
      >
        <div
          class="cell col"
        />
        <div
          class="cell col"
        />
        <div
          class="cell col"
        />
        <div
          class="cell col"
        />
      </div>
    </div>
  </div>
</div>
`;
